src/bosh-softlayer-cpi/action/create_vm.go

```diff
@@ -77,10 +77,18 @@ func (a CreateVMAction) Run(agentID string, stemcellCID StemcellCID, cloudProps
 func (a CreateVMAction) updateCloudProperties(cloudProps *VMCloudProperties) {
 	a.vmCloudProperties = cloudProps
 
-	if len(cloudProps.BoshIp) == 0 || cloudProps.Baremetal {
-		a.vmCloudProperties.VmNamePrefix = cloudProps.VmNamePrefix
+	if cloudProps.NotDeployedByDirector {
+		if len(cloudProps.Hostname) == 0 {
+			a.vmCloudProperties.VmNamePrefix = cloudProps.VmNamePrefix
+		} else {
+			a.vmCloudProperties.VmNamePrefix = cloudProps.Hostname
+		}
 	} else {
-		a.vmCloudProperties.VmNamePrefix = cloudProps.VmNamePrefix + TimeStampForTime(time.Now().UTC())
+		if len(cloudProps.Hostname) == 0 {
```

Comment:
I feel like lines 81-85 should be in private method with a param of optional timestamp... This avoids duplicating that code and make the if/else/if/else logic a bit better.